README.md

```diff
@@ -1,13 +1,137 @@
-erl_optics
-=====
+# erl_optics
 
-An OTP library
+An OTP application for [optics](https://github.com/RAttab/optics).
 
-### Requirements
+## Requirements
+-----
 
 * Erlang OTP >= 21
+* [optics](https://github.com/RAttab/optics) v1.1.10
 
-Build
+## Building
 -----
-
     $ rebar3 compile
+
+## Usage
+-----
+
+For now, the only working mode is Carbon.
+
+need env :
+```
+  {erl_optics, [
+      {hostname, "https://my-carbon-host.com"},
+      {port, 2003},
+      {mode, carbon},
+      {interval, 10000}
+  ]}
+```
+
+To start the application:
+```
+  erl_optics_app:start(),
+```
+
+To start the optics polling:
+```
+  erl_optics:start_optics(<<"my_namespace">>),
+```
+or
+```
+  erl_optics:start_optics(<<"my_namespace">>, Lenses),
+```
+
+The namespace will be prepending the name of every metric, for
+example: `"optics_namespace.my_counter"`
+
+The only lenses that require initial declaration are histo and
+quantile. If they are unused, they can be omitted, otherwise, they
+must be declared like so:
+
+```
+  Lenses = [erl_optics_lens:histo(<<"histo_name">>, [10, 20, 30, 40]),
+      erl_optics_lens:quantile(<<"quantile_name">>, 0.50, 10.0, 0.5)],
+  erl_optics:start_optics(<<"optics_namespace">>, Lenses).
+```
+
+## Lenses
+
+### counter
+A simple counter, polling will give the total increment value.
+Increment defaults to 1.
+```
+  erl_optics:counter_inc(<<"my_counter_name">>),
+```
+or
+```
+  erl_optics:counter_inc(<<"my_counter_name">, 10),
+```
+
+### gauge
+Will keep the last value given to it.
```

Comment:
I'm not sure I understand; if I don't produce a new value for a gauge, it's as if I keep sending the last value that I sent?

Reply:
Exactly. Useful for things like states, where you could for example have a gauge showing 0 when a switch is off or 1 when it's on.